src/software/ai/hl/stp/tactic/goalie/goalie_tactic.cpp

```diff
@@ -0,0 +1,63 @@
+#include "software/ai/hl/stp/tactic/goalie/goalie_tactic.h"
+
+#include "software/ai/evaluation/calc_best_shot.h"
+#include "software/ai/hl/stp/action/stop_action.h"  // TODO (#1888): remove this dependency
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/point.h"
+
+GoalieTactic::GoalieTactic(std::shared_ptr<const GoalieTacticConfig> goalie_tactic_config)
+    : Tactic(true,
+             {RobotCapability::Move, RobotCapability::Dribble, RobotCapability::Chip}),
+      fsm(),
+      goalie_tactic_config(goalie_tactic_config),
+      control_params{
+          GoalieFSM::ControlParams{.goalie_tactic_config = goalie_tactic_config}}
+{
+}
+
+void GoalieTactic::updateWorldParams(const World &world) {}
+
+void GoalieTactic::updateControlParams(
+    std::shared_ptr<const GoalieTacticConfig> updated_goalie_tactic_config)
+{
+    // Update the control parameters stored by this Tactic
+    control_params.goalie_tactic_config = updated_goalie_tactic_config;
+}
+
+double GoalieTactic::calculateRobotCost(const Robot &robot, const World &world) const
+{
+    // We don't prefer any particular robot to be the goalie, as there should only
+    // ever be one robot that can act as the goalie
+    return 0.5;
```

Comment:
Just curious why is 0.5 the chosen number if we don't prefer any particular robot?

Reply:
This was already here when I started to move things around, but I think it has something to do with how the goalie is usually just assigned for the whole game and not changing on the fly due to the in-game scenario? So robot cost wouldn't apply as much in the goalie's case, and 0.5 would be the neutral value... I may have understood this wrong though so maybe @jonathanlew can help me out with explaining this? :)

Reply:
Cost is never used for anything for the goalie so it really doesn't matter. A constant cost, i.e. not dependent on which robot, means every robot has the same cost and is equally likely to be assigned (if we used the cost in assignment)